cpp/src/table/row_operators.cu

```diff
@@ -301,6 +302,16 @@ void check_eq_compatibility(table_view const& input)
   }
 }
 
+void check_shape_compatibility(table_view const& lhs, table_view const& rhs)
```

Comment:
I belive that you have to call this recursively on the children.

Reply:
I believe this is already handling recursion. See previous comment https://github.com/rapidsai/cudf/pull/10730#discussion_r874054836 and the implementation of `column_types_equal`: https://github.com/rapidsai/cudf/blob/09b704552f2cdb4b1cbfe50a11c350fa4db1588b/cpp/src/utilities/type_checks.cpp#L66